File: 2026-01/src/day_01/boj_10845/Main.cpp
#include <bits/stdc++.h>
using namespace std;

const int MAX = 10000;
int q[MAX];
int s = 0, e = 0;

void push(int x) {
    q[e++] = x;
}

void pop() {
    s++;
}

int size() {
    return e - s;
}

bool empty() {
    return s == e;
}

int front() {
    return q[s];
}

int back() {
    return q[e - 1];
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;

    while (n--) {
        string q;
        cin >> q;

        if (q == "push") {
            int x;
            cin >> x;
            push(x);
        } else if (q == "pop") {
            if (empty()) {
                cout << -1 << '\n';
            } else {
                cout << front() << '\n';
                pop();
            }
        } else if (q == "size") {
            cout << size() << '\n';
        } else if (q == "empty") {
            cout << empty() << '\n';
        } else if (q == "front") {
            if (empty()) {
                cout << -1 << '\n';
            } else {
                cout << front() << '\n';
            }
        } else {
            if (empty()) {
                cout << -1 << '\n';
            } else {
                cout << back() << '\n';
            }
        }
    }
}

File: 2026-01/src/day_01/boj_10845/Main.java
package day_01.boj_10845;

import java.io.*;
import java.util.*;

public class Main {

    static final int MAX = 10000;
    static int[] queue = new int[MAX];
    static int front = 0, rear = 0;

    static void push(int x) {
        queue[rear++] = x;
    }

    static int pop() {
        return queue[front++];
    }

    static int size() {
        return rear - front;
    }

    static boolean empty() {
        return front == rear;
    }

    static int front() {
        return queue[front];
    }

    static int back() {
        return queue[rear - 1];
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;

        int n = Integer.parseInt(br.readLine());
        while (n-- > 0) {
            st = new StringTokenizer(br.readLine());
            String q = st.nextToken();

            if (q.equals("push")) {
                int x = Integer.parseInt(st.nextToken());
                push(x);
            } else if (q.equals("pop")) {
                if (empty()) {
                    sb.append("-1\n");
                } else {
                    sb.append(pop()).append("\n");
                }
            } else if (q.equals("size")) {
                sb.append(size()).append("\n");
            } else if (q.equals("empty")) {
                if (empty()) {
                    sb.append("1\n");
                } else {
                    sb.append("0\n");
                }
            } else if (q.equals("front")) {
                if (empty()) {
                    sb.append("-1\n");
                } else {
                    sb.append(front()).append("\n");
                }
            } else {
                if (empty()) {
                    sb.append("-1\n");
                } else {
                    sb.append(back()).append("\n");
                }
            }
        }

        System.out.println(sb);
    }
}
